File: src/main/java/org/homework20250105/java/animal/Dog.java
package org.homework20250105.java.animal;

import org.homework20250105.java.animal.base.Animal;
import org.homework20250105.java.animal.base.ManyFeet;

public class Dog extends Animal implements ManyFeet {
    public Dog(String name, Integer age) {
        super(name, age, AnimalEnum.DOG);
    }

    @Override
    public void say() {
        System.out.println("Woof");
    }

    @Override
    public int getFeet() {
        return 4;
    }
}

File: src/main/java/org/homework20250105/kotlin/animal/Dog.kt
package org.homework20250105.kotlin.animal

import org.homework20250105.kotlin.animal.base.Animal
import org.homework20250105.kotlin.animal.base.ManyFeet

class Dog(name: String, age: Int?) : Animal(name, age, AnimalEnum.DOG), ManyFeet {
    override fun say() {
        println("Woof")
    }

    override fun getFeet(): Int {
        return 4
    }
}
